### PR TITLE
Topic remove pusher none

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- remove `ParticlePusherNone` (not needed since #1226)
- remove documentation for `pusher::None`
- remove all includes of `ParticlePusherNone.hpp`
- add `PARAM_ENABLEPUSHER` to `SingleParticleTest`  (allow tests without any defined pusher)

Please do not merge before 
- [x] #1226 
- [x] #1205
- [x] #1377

### CODE REVIEW

**Comment by ax3l** [comment]
@psychocoderHPC pls rebase me

**Comment by psychocoderHPC** [comment]
rebased

**Comment by ax3l** [comment]
`SingleParticleTest` still seem to use `None`

**Comment by psychocoderHPC** [comment]
ready for review

File: examples/Bunch/include/simulation_defines/param/species.param [hunk 1]
```diff
@@ -73,7 +73,6 @@ typedef currentSolver::Esirkepov<UsedParticleShape> UsedParticleCurrentSolver;
  * - particles::pusher::Boris : standard boris pusher
  *
  * For development purposes: ---------------------------------------------------
- * - particles::pusher::None : no particle is pushed
  * - particles::pusher::Axel : a pusher developed at HZDR during 2011 (testing)
```
Comment (erikzenker) [inline examples/Bunch/include/simulation_defines/param/species.param]: Are pusher usually named by first or last name of the inventor ?

Comment (ax3l) [inline examples/Bunch/include/simulation_defines/param/species.param]: by inventor's choice, and it was Axl, not Axel.

**Comment by psychocoderHPC** [comment]
ready for merge

File: examples/SingleParticleTest/cmakeFlags [hunk 2]
```diff
@@ -36,17 +36,14 @@ flags[1]="-DCUDA_ARCH=sm_20 -DPARAM_OVERWRITES:LIST=-DPARAM_DIMENSION=DIM2"
 # Vay Pusher
 flags[2]="-DCUDA_ARCH=sm_20 -DPARAM_OVERWRITES:LIST=-DPARAM_PARTICLEPUSHER=Vay"
 flags[3]="-DCUDA_ARCH=sm_20 -DPARAM_OVERWRITES:LIST=-DPARAM_PARTICLEPUSHER=Vay;-DPARAM_DIMENSION=DIM2"
-# None Pusher
-flags[4]="-DCUDA_ARCH=sm_20 -DPARAM_OVERWRITES:LIST=-DPARAM_PARTICLEPUSHER=None"
-flags[5]="-DCUDA_ARCH=sm_20 -DPARAM_OVERWRITES:LIST=-DPARAM_PARTICLEPUSHER=None;-DPARAM_DIMENSION=DIM2"
```
Comment (ax3l) [inline examples/SingleParticleTest/cmakeFlags]: having a test compile with particles that explicitly have "no pusher" in the new design would still be useful. pls add one again.

Comment (ax3l) [inline examples/SingleParticleTest/cmakeFlags]: -> `git cp speciesDefinition.param` from defaults needed for that

**Comment by ax3l** [comment]
@psychocoderHPC needs a minor update to be merged :)

**Comment by ax3l** [comment]
@psychocoderHPC pls edit the minor comment and rebase

**Comment by psychocoderHPC** [comment]
is rebased and ready to merge

File: examples/SingleParticleTest/cmakeFlags (hunk 2 above)
Comment (psychocoderHPC) [inline examples/SingleParticleTest/cmakeFlags]: The tests for disabled pusher are now in

File: examples/SingleParticleTest/include/simulation_defines/param/speciesDefinition.param [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+AttributMomentum_mt1,
+AttributRadiationFlag
+>::type DefaultAttributesSeq;
+
+/*########################### end particle attributes ########################*/
+
+/*########################### define species #################################*/
+
+/* enable pusher by default if `PARAM_ENABLEPUSHER` is not defined in `cmakeFlags` */
+#ifndef PARAM_ENABLEPUSHER
```
Comment (ax3l) [inline examples/SingleParticleTest/include/simulation_defines/param/speciesDefinition.param]: style:
```C++
#ifndef PARAM_ENABLEPUSHER
#    define PARAM_ENABLEPUSHER 1
#endif

**Comment by psychocoderHPC** [comment]
updated

**Pull request merged**